cassiopeia/simulator/DataSimulator.py

```diff
@@ -0,0 +1,19 @@
+"""
+Abstract class DataSimulator, for overlaying data onto a CassiopeiaTree.
+
+All data simulators are derived classes of this abstract class, and at a minimum
+implement a method called `overlay_data`.
+"""
+import abc
+from cassiopeia.data import CassiopeiaTree
+
+
+class DataSimulator(abc.ABC):
+    """
+    DataSimulator is an abstract class that all data overlayers derive from.
```

Comment:
Let's expand a bit on the intended use case of the `DataSimulator` -- how should users use this, and how should it be extended? Are there certain things it should _not_ be used for?

Reply:
We should probably discuss whether we should have this class at all right now.

Reply:
Your updated documentation is great -- this was what I was looking for. Just for the sake of posterity, to recap our conversation over slack, we'd like to have this `DataSimulator` abstract class because we'd like to extend the simulator to simulate spatial location, tissue identity, transcriptomic state, etc.